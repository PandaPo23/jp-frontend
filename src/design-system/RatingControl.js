import PropTypes from 'prop-types';
import React from 'react';
import Rating from 'react-rating';

import Icon from './Icon';

const RatingControl = ({
  size,
  emptyIcon,
  fullIcon,
  color,
  fullColor,
  ...props
}) => (
  <Rating
    {...props}
    emptySymbol={<Icon size={size} name={emptyIcon} color={color} />}
    fullSymbol={<Icon size={size} name={fullIcon} color={fullColor} />}
  />
);

RatingControl.displayName = 'RatingControl';

RatingControl.propTypes = {
  size: PropTypes.number,
  emptyIcon: PropTypes.string,
  fullIcon: PropTypes.string,
  color: PropTypes.string,
  fullColor: PropTypes.string,
  ...Rating.propTypes,
};

RatingControl.defaultProps = {
  size: 28,
  emptyIcon: 'star-border',
  fullIcon: 'star',
  color: 'border',
  fullColor: 'primary',
  ...Rating.defaultProps,
};

export default RatingControl;
