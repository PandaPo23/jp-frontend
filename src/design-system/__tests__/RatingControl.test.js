import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RatingControl from '../RatingControl';

describe('RatingControl component', () => {
  it('renders with specified color state correctly', () => {
    const localProps = {
      color: 'orange',
      fullColor: 'pink',
    };
    const wrapper = shallow(<RatingControl {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with custom icons correctly', () => {
    const localProps = {
      fractions: 2,
      emptyIcon: 'empty-circle',
      fullIcon: 'full-circle',
      size: 18,
    };
    const wrapper = shallow(<RatingControl {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
