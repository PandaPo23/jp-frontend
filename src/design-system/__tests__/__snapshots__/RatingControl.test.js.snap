// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingControl component renders with custom icons correctly 1`] = `
<RatingAPILayer
  direction="ltr"
  emptySymbol={
    <ForwardRef(render)
      color="border"
      name="empty-circle"
      size={18}
    />
  }
  fractions={2}
  fullSymbol={
    <ForwardRef(render)
      color="primary"
      name="full-circle"
      size={18}
    />
  }
  onChange={[Function]}
  onClick={[Function]}
  onHover={[Function]}
  placeholderSymbol={
    Object {
      "backgroundColor": "red",
      "border": "5px double white",
      "borderRadius": "50%",
      "display": "inline-block",
      "height": 30,
      "width": 30,
    }
  }
  quiet={false}
  readonly={false}
  start={0}
  step={1}
  stop={5}
/>
`;

exports[`RatingControl component renders with specified color state correctly 1`] = `
<RatingAPILayer
  direction="ltr"
  emptySymbol={
    <ForwardRef(render)
      color="orange"
      name="star-border"
      size={28}
    />
  }
  fractions={1}
  fullSymbol={
    <ForwardRef(render)
      color="pink"
      name="star"
      size={28}
    />
  }
  onChange={[Function]}
  onClick={[Function]}
  onHover={[Function]}
  placeholderSymbol={
    Object {
      "backgroundColor": "red",
      "border": "5px double white",
      "borderRadius": "50%",
      "display": "inline-block",
      "height": 30,
      "width": 30,
    }
  }
  quiet={false}
  readonly={false}
  start={0}
  step={1}
  stop={5}
/>
`;
